server/api/middlewares/collectionIndexing.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+import fractionalIndex from "fractional-index";
+import { type Context } from "koa";
+import { Collection } from "../../models";
+export default function collectionIndexing() {
+  return async function collectionIndexingMiddleware(
+    ctx: Context,
+    next: () => Promise<*>
+  ) {
+    const teamId = ctx.state.user.teamId;
+    const collections = await Collection.findAll({
```

Comment:
The problem with this approach is you're pulling all the collections from the db whether they're needed or not on every request that uses the middleware. Consider we already load collections inside of `collections.list` why not perform this as a one-off when those collections are loaded just check to see if first has a `null` index property?

Reply:
I think we can use the users collections from the existing query as a "hint" to whether the migration needs to happen if the first index is `null`, and then in only that case we can query _all_ collections and fix them.